Core/GDCore/Events/CodeGeneration/EventsCodeGenerator.cpp

```diff
@@ -60,14 +60,36 @@ gd::String EventsCodeGenerator::GenerateRelationalOperatorCall(
             1);  // Relational operator contains quote which must be removed.
 
   gd::String rhs = arguments[relationalOperatorIndex + 1];
+  bool caseInsensitive = arguments[relationalOperatorIndex + 2] == "true";
+  gd::String caseInsensitiveStr = caseInsensitive ? "i" : "";
+
   gd::String argumentsStr;
   for (std::size_t i = startFromArgument; i < arguments.size(); ++i) {
-    if (i != relationalOperatorIndex && i != relationalOperatorIndex + 1) {
+    if (i != relationalOperatorIndex && i != relationalOperatorIndex + 1 && i != relationalOperatorIndex + 2) {
       if (!argumentsStr.empty()) argumentsStr += ", ";
       argumentsStr += arguments[i];
     }
   }
 
+  // Check type of Relational Operator
+  if (relationalOperator == "startsWith") {
+    return rhs + " != '' && " + callStartString + "(" + argumentsStr +
+        ").match(new RegExp('^' + " + rhs + ", '" + caseInsensitiveStr + "')) !== null";
+  } else if (relationalOperator == "endsWith") {
+    return rhs + " != '' && " + callStartString + "(" + argumentsStr +
+        ").match(new RegExp(" + rhs + " + '$', '" + caseInsensitiveStr + "')) !== null";
+  } else if (relationalOperator == "contains") {
+    return rhs + " != '' && " + callStartString + "(" + argumentsStr +
+        ").match(new RegExp(" + rhs + ", '" + caseInsensitiveStr + "')) !== null";
+  } else if (caseInsensitive && relationalOperator == "==") {
+    return callStartString + "(" + argumentsStr +
+        ").match(new RegExp('^' + " + rhs + " + '$', '" + caseInsensitiveStr + "')) !== null";
+  } else if (caseInsensitive && relationalOperator == "!=") {
+    return callStartString + "(" + argumentsStr +
+        ").match(new RegExp('^' + " + rhs + " + '$', '" + caseInsensitiveStr + "')) === null";
+  }
+
+  // Standard Relational Operator (==, !=)

```

Comment:
Does case sensitivity is handled for `==` and `!=`?

Reply:
HI @D8H , Yes, thats it. The reasons are:

1. It's backward compatible.
2. The method `GenerateRelationalOperatorCall()` is not exclusive for "Compare texts" conditions, its algo used in other conditions that don't include directly text. So in this way we are sure that conditions continue working.